rio_tiler_pds/landsat/aws/landsat_collection2.py

```diff
@@ -100,37 +117,34 @@ class LandsatC2L2Reader(MultiBandReader):
 
     _scheme: str = "s3"
     _hostname: str = "usgs-landsat"
-    _prefix: str = "collection02/level-{_processingLevelNum}/standard/{_sensor}/{acquisitionYear}/{path}/{row}/{scene}/{scene}"
+    _prefix: str = "collection02/level-{_processingLevelNum}/standard/{sensor_name}/{acquisitionYear}/{path}/{row}/{scene}/{scene}"
 
     def __attrs_post_init__(self):
         """Fetch productInfo and get bounds."""
         self.scene_params = sceneid_parser(self.sceneid)
 
         processing_level = self.scene_params["processingCorrectionLevel"]
-        sensor = self.scene_params["_sensor"]
+        sensor_name = self.scene_params["sensor_name"]
 
-        # For collection 2, level 2, ETM and TM have the same bands
         if processing_level == "L2SR":
-            if sensor == "oli-tirs":
+            if sensor_name in ["oli-tirs", "oli"]:
                 self.bands = OLI_TIRS_SR_BANDS
-            elif sensor == "tm":
-                self.bands = TM_SR_BANDS
-            elif sensor == "etm":
+            elif sensor_name in ["tm", "etm"]:
                 self.bands = TM_SR_BANDS
         else:
-            if sensor == "oli-tirs":
+            if sensor_name == "oli-tirs":
                 self.bands = OLI_TIRS_SR_BANDS + OLI_TIRS_ST_BANDS
-            elif sensor == "tm":
-                self.bands = TM_SR_BANDS + TM_ST_BANDS
-            elif sensor == "etm":
+            elif sensor_name in ["tm", "etm"]:
                 self.bands = TM_SR_BANDS + TM_ST_BANDS
+
         # TODO: add separate level 1 reader with TOA reflectances
         # elif self.scene_params['processingCorrectionLevel'] in ['L1TP', 'L1GT', 'L1GS']:
         #     self.bands = DEFAULT_L8L1_BANDS
 
-        self._get_geometry()
+        self.bounds = self.get_geometry()
```

Comment:
by doing this, a user could subclass the reader and implement their own `get_geometry` method